src/vendingMachine.js

```diff
@@ -0,0 +1,217 @@
+    for(let i = 0; i < insertedCoinsArray.length; i++){
+      if(insertedCoinsArray[i].stock > 0){
+        total += (insertedCoinsArray[i].stock*insertedCoinsArray[i].value);
+      }
+    }
+    return total;
+  }
+  dispenseInventory(product, insertedCoins){
+    const afterDispensedInventory = inventory;
+    const afterDispensedCoins = coin;
+    const insertedCoinsArray = insertedCoins;
+
+    for(let i = 0; i < afterDispensedInventory.length; i++){
+      if (afterDispensedInventory[i].name === product){
+        if(afterDispensedInventory[i].stock > 0){
+          afterDispensedInventory[i].stock += -1;
+          const productPrice = afterDispensedInventory[i].price
+        }else{
+          return 'out of that product, please choose another';
+        }
+      }
+    }
+    for (let i = 0; i < afterDispensedCoins.length; i++){
+      if(insertedCoins[i].stock > 0){
+        afterDispensedCoins[i].stock += 1;
+      }
+    }
+    return afterDispensedInventory;
+  }
+  returnChange(productValue, insertedCoins, coin){
+    console.log('insertedCoins', insertedCoins)
+    const insertedTotal = this.currentTotal(insertedCoins);
+    const machineCoins = coin;
+    let changeCounter = insertedTotal - productValue;
+    console.log('changeCounter', changeCounter);
+    if(changeCounter < 0){
+      return `not enough change, insert ${-changeCounter} more`
+    }
+    let returnedCoinsArray= returnedCoins;
+    while(changeCounter > 0){
+      if(changeCounter >= 2){
+        for(let i=0; i < machineCoins.length; i++){
+          if(machineCoins[i].type === 'tooney'){
+            machineCoins[i].stock -= 1;
+            changeCounter -= machineCoins[i].value;
+            returnedCoinsArray[i].stock += 1;
+          }
+        }
+      }else if( changeCounter >= 1){
+        for(let i=0; i<machineCoins.length;i++){
+          if(machineCoins[i].type === 'looney'){
+            machineCoins[i].stock -= 1;
+            changeCounter -= machineCoins[i].value;
+            returnedCoinsArray[i].stock += 1;
+          }
+        }
+      }else if( changeCounter >= .25){
+        for(let i=0; i<machineCoins.length;i++){
+          if(machineCoins[i].type === 'quarter'){
+            machineCoins[i].stock -= 1;
+            changeCounter -= machineCoins[i].value;
+            returnedCoinsArray[i].stock += 1;
+          }
+        }
+      }else if( changeCounter >= .10){
+        for(let i=0; i<machineCoins.length;i++){
+          if(machineCoins[i].type === 'dime'){
+            machineCoins[i].stock += -1;
+            changeCounter -= machineCoins[i].value;
+            returnedCoinsArray[i].stock += 1;
+          }
+        }
+      }else if( changeCounter >= .05){
+        for(let i=0; i< machineCoins.length;i++){
+          if(machineCoins[i].type === 'nickel'){
+            machineCoins[i].stock -= 1;
+            changeCounter -= machineCoins[i].value;
+            returnedCoinsArray[i].stock += 1;
+          }
+        }
+      }
+    }
```

Comment:
I know this is a bit of a mess, but you should be pretty happy with this code. It shows that you're thinking one of the most difficult parts of this question through. This would be a really good sign to me if I was thinking about hiring you.